packages/calcite-components/src/components/modal/modal.tsx

```diff
@@ -557,7 +557,11 @@ export class Modal
     }
 
     if (this.beforeClose) {
-      await this.beforeClose(this.el);
+      try {
+        await this.beforeClose(this.el);
+      } catch (_error) {
+        return;
```

Comment:
Can you add a `// close prevented` or similar one-liner comment to the catch block about this intentionally returning because closing was canceled?